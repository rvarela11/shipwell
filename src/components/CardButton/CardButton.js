/* eslint-disable camelcase */
// @vendors
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

// @material-ui
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';

// Setting material-ui classes
const styles = () => ({
    card_button: {
        color: 'white'
    }
});

const CardButton = (props) => {
    const { classes, label, isCardButtonDisabled } = props;
    return (
        <Link to="/shipwell-results" className={`link ${isCardButtonDisabled ? 'disable-link' : ''}`}>
            <Button
                className={classes.card_button}
                color="primary"
                disabled={isCardButtonDisabled}
                size="large"
                variant="contained"
            >
                {label}
            </Button>
        </Link>
    );
};

CardButton.propTypes = {
    classes: PropTypes.object.isRequired,
    label: PropTypes.string.isRequired,
    isCardButtonDisabled: PropTypes.bool.isRequired
};

export default withStyles(styles)(CardButton);
