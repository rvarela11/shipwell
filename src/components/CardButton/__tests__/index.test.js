// @vendors
import React from 'react';
import { shallow } from 'enzyme';

// @components
import CardButton from '../CardButton';

describe('CardButton', () => {
    it('renders CardButton without crashing', () => {
        const wrapper = shallow(
            <CardButton
                classes={{}}
                label=""
                isCardButtonDisabled
                path=""
            />
        ).dive();
        expect(wrapper).toMatchSnapshot();
    });
});
