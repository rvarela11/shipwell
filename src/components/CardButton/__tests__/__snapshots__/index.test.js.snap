// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardButton renders CardButton without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardButton
    classes={
      Object {
        "card_button": "CardButton-card_button-1",
      }
    }
    isCardButtonDisabled={true}
    label=""
    path=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Button)
        className="CardButton-card_button-1"
        color="primary"
        disabled={true}
        size="large"
        variant="contained"
      >
        
      </WithStyles(Button)>,
      "className": "link disable-link",
      "replace": false,
      "to": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "",
        "className": "CardButton-card_button-1",
        "color": "primary",
        "disabled": true,
        "size": "large",
        "variant": "contained",
      },
      "ref": null,
      "rendered": "",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Button)
          className="CardButton-card_button-1"
          color="primary"
          disabled={true}
          size="large"
          variant="contained"
        >
          
        </WithStyles(Button)>,
        "className": "link disable-link",
        "replace": false,
        "to": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
          "className": "CardButton-card_button-1",
          "color": "primary",
          "disabled": true,
          "size": "large",
          "variant": "contained",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
